app/state.go

```diff
@@ -1835,3 +1847,207 @@ func setBrokenArtifactFlag(store store.Store, artName string) {
+	log.Debugf("controlMapState action: %s", action)
+	switch action {
+	case "continue":
+		return c.wrappedState, false
+	case "pause":
+		log.Infof("Update Control: Pausing before entering %s state", c.wrappedState.Id())
+		log.Debugf("Reporting update status: %s", c.pauseName(c.wrappedState.Transition()))
+		if !ctx.pauseReported[c.wrappedState.Id().String()] {
+			ctx.pauseReported[c.wrappedState.Id().String()] = true
+			merr := controller.ReportUpdateStatus(
+				c.wrappedState.Update(),
+				c.pauseName(c.wrappedState.Transition()),
+			)
+			if merr != nil && merr.IsFatal() {
+				return c.wrappedState.HandleError(ctx, controller, merr)
+			}
+		}
+		return NewControlMapPauseState(c.wrappedState), false
+	case "fail":
+		log.Infof("Update Control: Forced update failure in %s state", c.wrappedState.Id())
+		return c.wrappedState.HandleError(ctx, controller,
+			NewTransientError(errors.New("Forced a failed update")))
+	default:
+		log.Warnf("Unknown Action: %s, continuing", action)
+		return c.wrappedState, false
+	}
+}
+
+type fetchControlMapState struct {
+	baseState
+	update       *datastore.UpdateInfo
+	wrappedState UpdateState
+}
+
+func NewFetchControlMapState(wrappedState UpdateState) State {
+	return &fetchControlMapState{
+		baseState: baseState{
+			id: datastore.MenderStateFetchUpdateControl,
+			t:  ToNone,
+		},
+		wrappedState: wrappedState,
+	}
+}
+
+func (c *fetchControlMapState) Handle(ctx *StateContext, controller Controller) (State, bool) {
+
+	log.Debugf("Handle fetchControlMap state")
+
+	// The update control maps are magically updated in here
+	_, err := controller.CheckUpdate()
+
+	if err != nil {
+		if errors.Is(err, client.ErrNoDeploymentAvailable) {
+			return c.wrappedState.HandleError(ctx, controller,
+				NewTransientError(errors.New("The deployment was aborted from the server")))
+		}
+
+		log.Errorf("Update control map check failed: %s, retrying...", err)
+		return NewFetchRetryControlMapState(c.wrappedState), false
+	}
+
+	return NewControlMapState(c.wrappedState), false
+}
+
+type fetchRetryControlMapState struct {
+	waitState
+	wrappedState UpdateState
+	retries      int
+}
+
+func NewFetchRetryControlMapState(wrappedState UpdateState) State {
+	return &fetchRetryControlMapState{
+		waitState: waitState{
+			baseState: baseState{
+				id: datastore.MenderStateFetchRetryUpdateControl,
+				t:  ToNone,
+			},
+		},
+		wrappedState: wrappedState,
+	}
+}
+
+func (f *fetchRetryControlMapState) Handle(ctx *StateContext, c Controller) (State, bool) {
+
+	log.Debugf("Handle fetch update control retry state")
+
+	intvl, err := client.GetExponentialBackoffTime(f.retries, c.GetUpdatePollInterval())
+	if err != nil {
+		return NewUpdateErrorState(
+			NewTransientError(err), f.wrappedState.Update()), false
+	}
+
+	f.retries++
+
+	log.Infof("Wait %v before next update control map fetch/update attempt", intvl)
+	return f.Wait(
+		NewFetchControlMapState(f.wrappedState),
+		f,
+		intvl,
+		ctx.WakeupChan,
```

Comment:
One option here, is to wake on updates to the control maps, and return to the `QueryAndUpdate` loop